### ISSUE TITLE
Jar resolution fails when Windows file paths have spaces

### ISSUE THREAD

**Issue by TimothyDJewell** [thheller/shadow-cljs#732]
I made a foolish decision several years ago and put a space in my Windows user name, which then meant my user directory also has a space.  Unfortunately, this tends to be a failure point when using software, so I at least know where to look.
Recently I decided to try playing with re-frame, so I ran `lein new re-frame start-small`, followed by `lein deps`.  Neither had issue.   Then I tried running `lein dev` and got:

```
~\Documents\ClojureScript\start-small> lein dev
lein-shadow - reading NPM dependencies from C:\Users\Timothy Jewell\Documents\ClojureScript\start-small\src\cljs\deps.cljs
lein-shadow - running: cmd /C npm --version
lein-shadow - NPM version 5.6.0
lein-shadow - running: cmd /C npm install --save-dev --save-exact shadow-cljs@2.9.3
lein-shadow - NPM install successful + shadow-cljs@2.9.3added 99 packages in 6.752s
lein-shadow - running shadow-cljs...
Retrieving nrepl/nrepl/0.7.0/nrepl-0.7.0.jar from clojars
running: cmd /C npm install --save --save-exact react@16.13.0 react-dom@16.13.0
npm WARN start-small@ No repository field.
npm WARN start-small@ No license field.

+ react@16.13.0
+ react-dom@16.13.0
added 7 packages in 2.163s
shadow-cljs - HTTP server available at http://localhost:8280
shadow-cljs - server version: 2.9.3 running at http://localhost:9630
shadow-cljs - nREPL server started on port 8777
shadow-cljs - watching build :app
[:app] Configuring build.
[:app] Compiling ...
[:app] Build failure:
expected to find jar file but didn't
{:jar-path "/C:/Users/Timothy%20Jewell/.m2/repository/binaryage/devtools/1.0.0/devtools-1.0.0.jar"}
ExceptionInfo: expected to find jar file but didn't
        shadow.build.classpath/get-jar-info* (classpath.clj:463)
        shadow.build.classpath/get-jar-info* (classpath.clj:460)
        clojure.core/apply (core.clj:665)
        clojure.core/memoize/fn--6877 (core.clj:6353)
        shadow.build.classpath/get-jar-info-for-url (classpath.clj:480)
        shadow.build.classpath/get-jar-info-for-url (classpath.clj:472)
        shadow.build.classpath/make-jar-resource (classpath.clj:484)
        shadow.build.classpath/make-jar-resource (classpath.clj:482)
        shadow.build.classpath/find-resource-by-name (classpath.clj:1132)
        shadow.build.classpath/find-resource-by-name (classpath.clj:1120)
        shadow.build.classpath/find-resource-for-provide (classpath.clj:1144)
        shadow.build.classpath/find-resource-for-provide (classpath.clj:1139)
        shadow.build.resolve/find-resource-for-symbol* (resolve.clj:410)
        shadow.build.resolve/find-resource-for-symbol* (resolve.clj:400)
        shadow.build.resolve/find-resource-for-symbol (resolve.clj:451)
        shadow.build.resolve/find-resource-for-symbol (resolve.clj:445)
        shadow.build.resolve/resolve-symbol-require (resolve.clj:505)
        shadow.build.resolve/resolve-symbol-require (resolve.clj:501)
        shadow.build.resolve/resolve-require (resolve.clj:552)
        shadow.build.resolve/resolve-require (resolve.clj:548)
        shadow.build.resolve/resolve-entry (resolve.clj:562)
        shadow.build.resolve/resolve-entry (resolve.clj:561)
        clojure.lang.PersistentVector.reduce (PersistentVector.java:343)
        clojure.core/reduce (core.clj:6827)
        clojure.core/reduce (core.clj:6810)
        shadow.cljs.util/reduce-> (util.clj:47)
        shadow.cljs.util/reduce-> (util.clj:46)
        shadow.build.resolve/resolve-entries (resolve.clj:573)
        shadow.build.resolve/resolve-entries (resolve.clj:564)
        shadow.build.modules/resolve-module/fn--14063 (modules.clj:265)
        shadow.build.modules/resolve-module (modules.clj:261)
        shadow.build.modules/resolve-module (modules.clj:251)
        clojure.lang.PersistentVector.reduce (PersistentVector.java:343)
        clojure.core/reduce (core.clj:6827)
        clojure.core/reduce (core.clj:6810)
        shadow.build.modules/resolve-modules (modules.clj:271)
        shadow.build.modules/resolve-modules (modules.clj:270)
        shadow.build.modules/analyze (modules.clj:325)
        shadow.build.modules/analyze (modules.clj:316)
        shadow.build/resolve (build.clj:395)
        shadow.build/resolve (build.clj:389)
        shadow.build/compile (build.clj:403)
        shadow.build/compile (build.clj:397)
        shadow.cljs.devtools.server.worker.impl/build-compile (impl.clj:341)
        shadow.cljs.devtools.server.worker.impl/build-compile (impl.clj:323)
        shadow.cljs.devtools.server.worker.impl/eval15959/fn--15961 (impl.clj:775)
        clojure.lang.MultiFn.invoke (MultiFn.java:234)
        shadow.cljs.devtools.server.util/server-thread/fn--15565/fn--15566/fn--15574 (util.clj:285)
        shadow.cljs.devtools.server.util/server-thread/fn--15565/fn--15566 (util.clj:284)
        shadow.cljs.devtools.server.util/server-thread/fn--15565 (util.clj:257)
        java.lang.Thread.run (Thread.java:834)
```

My first thought was: "Does that path exist?" (translating %20 into a space), so I verified the path manually, and then I read through classpath.clj trying to figure out what was going on.  I started up the repl and ran tests on the variants I could think of:

```
user=> (def path "C:/Users/Timothy Jewell/.m2/repository/binaryage/devtools/1.0.0/devtools-1.0.0.jar")
user=> (let [file (clojure.java.io/file path)
  #_=>       exists (.exists file)
  #_=>       isFile (.isFile file)]
  #_=>   (str "Exists: " exists "; IsFile: " isFile "."))
"Exists: true; IsFile: true."
user=> (def path "C:/Users/Timothy%20Jewell/.m2/repository/binaryage/devtools/1.0.0/devtools-1.0.0.jar")
#'user/path
user=> (let [file (clojure.java.io/file path)
  #_=>       exists (.exists file)
  #_=>       isFile (.isFile file)]
  #_=>   (str "Exists: " exists "; IsFile: " isFile "."))
"Exists: false; IsFile: false."
user=> (def path "/C:/Users/Timothy%20Jewell/.m2/repository/binaryage/devtools/1.0.0/devtools-1.0.0.jar")
#'user/path
user=> (let [file (clojure.java.io/file path)
  #_=>       exists (.exists file)
  #_=>       isFile (.isFile file)]
  #_=>   (str "Exists: " exists "; IsFile: " isFile "."))
"Exists: false; IsFile: false."
user=> (def path "/C:/Users/Timothy Jewell/.m2/repository/binaryage/devtools/1.0.0/devtools-1.0.0.jar")
#'user/path
user=> (let [file (clojure.java.io/file path)
  #_=>       exists (.exists file)
  #_=>       isFile (.isFile file)]
  #_=>   (str "Exists: " exists "; IsFile: " isFile "."))
"Exists: true; IsFile: true."
```
This suggests the issue is that some code is URI encoding the space, and converting it to `%20`, which does not seem to work for Windows file paths.

**Comment by thheller** [comment]
Pretty sure this is fixed in the latest version. Please try `2.10.6`.

**Comment by TimothyDJewell** [comment]
It doesn't seem to be fixed in 2.10.6.  I manually changed all references to 2.9.3 to 2.10.6 and retried (`lein deps`, `lein dev`).  Even tried clearing the contents of .shadow-cljs\jar-manifest\ and letting the build retry, as well as 2.10.5.

Output was:
```
~\Documents\ClojureScript\start-small> lein dev
lein-shadow - reading NPM dependencies from C:\Users\Timothy Jewell\Documents\ClojureScript\start-small\src\cljs\deps.cljs
lein-shadow - running: cmd /C npm --version
lein-shadow - NPM version 5.6.0
lein-shadow - running: cmd /C npm install --save-dev --save-exact shadow-cljs@2.10.6
lein-shadow - NPM install successful + shadow-cljs@2.10.6updated 1 package in 2.909s
lein-shadow - running shadow-cljs...
shadow-cljs - HTTP server available at http://localhost:8280
shadow-cljs - server version: 2.10.6 running at http://localhost:9630
shadow-cljs - nREPL server started on port 8777
shadow-cljs - watching build :app
[:app] Configuring build.
[:app] Compiling ...
[:app] Build failure:
expected to find jar file but didn't
{:jar-path "/C:/Users/Timothy%20Jewell/.m2/repository/binaryage/devtools/1.0.0/devtools-1.0.0.jar"}
ExceptionInfo: expected to find jar file but didn't
        shadow.build.classpath/get-jar-info* (classpath.clj:465)
        shadow.build.classpath/get-jar-info* (classpath.clj:462)
        clojure.core/apply (core.clj:665)
        clojure.core/memoize/fn--6877 (core.clj:6353)
        shadow.build.classpath/get-jar-info-for-url (classpath.clj:482)
        shadow.build.classpath/get-jar-info-for-url (classpath.clj:474)
        shadow.build.classpath/make-jar-resource (classpath.clj:486)
        shadow.build.classpath/make-jar-resource (classpath.clj:484)
        shadow.build.classpath/find-resource-by-name (classpath.clj:1134)
        shadow.build.classpath/find-resource-by-name (classpath.clj:1122)
        shadow.build.classpath/find-resource-for-provide (classpath.clj:1146)
        shadow.build.classpath/find-resource-for-provide (classpath.clj:1141)
        shadow.build.resolve/find-resource-for-symbol* (resolve.clj:420)
        shadow.build.resolve/find-resource-for-symbol* (resolve.clj:410)
        shadow.build.resolve/find-resource-for-symbol (resolve.clj:461)
        shadow.build.resolve/find-resource-for-symbol (resolve.clj:455)
        shadow.build.resolve/resolve-symbol-require (resolve.clj:515)
        shadow.build.resolve/resolve-symbol-require (resolve.clj:511)
        shadow.build.resolve/resolve-require (resolve.clj:562)
        shadow.build.resolve/resolve-require (resolve.clj:558)
        shadow.build.resolve/resolve-entry (resolve.clj:572)
        shadow.build.resolve/resolve-entry (resolve.clj:571)
        clojure.lang.PersistentVector.reduce (PersistentVector.java:343)
        clojure.core/reduce (core.clj:6827)
        clojure.core/reduce (core.clj:6810)
        shadow.cljs.util/reduce-> (util.clj:47)
        shadow.cljs.util/reduce-> (util.clj:46)
        shadow.build.resolve/resolve-entries (resolve.clj:586)
        shadow.build.resolve/resolve-entries (resolve.clj:577)
        shadow.build.modules/resolve-module/fn--14174 (modules.clj:265)
        shadow.build.modules/resolve-module (modules.clj:261)
        shadow.build.modules/resolve-module (modules.clj:251)
        clojure.lang.PersistentVector.reduce (PersistentVector.java:343)
        clojure.core/reduce (core.clj:6827)
        clojure.core/reduce (core.clj:6810)
        shadow.build.modules/resolve-modules (modules.clj:271)
        shadow.build.modules/resolve-modules (modules.clj:270)
        shadow.build.modules/analyze (modules.clj:325)
        shadow.build.modules/analyze (modules.clj:316)
        shadow.build/resolve (build.clj:397)
        shadow.build/resolve (build.clj:391)
        shadow.build/compile (build.clj:429)
        shadow.build/compile (build.clj:422)
        shadow.cljs.devtools.server.worker.impl/build-compile (impl.clj:364)
        shadow.cljs.devtools.server.worker.impl/build-compile (impl.clj:345)
        shadow.cljs.devtools.server.worker.impl/eval15891/fn--15893 (impl.clj:440)
        clojure.lang.MultiFn.invoke (MultiFn.java:234)
        shadow.cljs.devtools.server.util/server-thread/fn--15653/fn--15654/fn--15662 (util.clj:285)
        shadow.cljs.devtools.server.util/server-thread/fn--15653/fn--15654 (util.clj:284)
        shadow.cljs.devtools.server.util/server-thread/fn--15653 (util.clj:257)
        java.lang.Thread.run (Thread.java:834)
```

**Comment by thheller** [comment]
Should be fixed in `2.10.7`.

**Comment by TimothyDJewell** [comment]
It is! Thank you so much!
